### PR TITLE
Add `access` permission to accounts

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds an `access` permission to accounts, which prohibits zkApp accounts from being accessed if the authorization is incorrect. This ensures that an 'empty update' to e.g. a token zkApp can't be used to pass unauthorized sub-calls.

Testing is still TODO; opening this to see if CI finds any glaring errors in existing tests.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by psteckler** [comment]
Do you want to add a test that exercises this permission?

**Review by mitschabaude — APPROVED** [review]
looks great!

**Review by bkase — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/mina_base/account.ml [hunk 1]
```diff
@@ -580,17 +580,37 @@ module Checked = struct
...
+      | None -> (
+          match to_ with
+          | `Send ->
+              Boolean.true_
+          | `Receive ->
+              Boolean.false_
+          | `Set_delegate ->
+              Boolean.true_
+          | `Access ->
+              failwith
```
Comment (bkase) [inline src/lib/mina_base/account.ml:597]: why put a crasher here? seems unsafe?

**Review by bkase — APPROVED** [review]

File: src/lib/mina_base/account.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/mina_base/account.ml:597]: There's no sensible default value for `access`, so this ensures that we call it correctly everywhere.

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**